### PR TITLE
[CDAP-17246] Fixes pipeline exported in 6.1.x CDAP is properly deployed in later versions

### PR DESCRIPTION
**Pull request by ajainarayanan**
**TL;DR** 
- Fixes pipeline studio to import pipelines exported in older version of CDAP (<6.2). 
- Separate plugin name from being used in DOM id attribute. Prevents modifying plugin name which could be used in other plugin properties (joiner)

JIRA: https://issues.cask.co/browse/CDAP-17246

**Longer Version**

The fix involves using a unique ID that will be used in DOM node. The reason we changed the name in
the first place was to have no spaces (or other spl characters) that are not supported by DOM id attribute.
Since the same stage name could be used in other plugin properties we no longer change the name during pipeline
imports.

We however have to update the stage name when copy/pasting nodes in the pipeline studio. We need to do this as
the stage name uniquely identifies the plugin in the pipeline DAG. This change makes sure stage names modified
while copy/pasting nodes to pipeline canvas is correspondingly updated in other plugins where the plugin
name is used.

There is still a small window where users using custom plugin in which none of the widgets are used (just plain textbox) but still references previous stages. The UI won't be catch this case as of now. Corresponding JIRA for this use case - https://issues.cask.co/browse/CDAP-17252

**UPDATE:**
We decided not to auto update plugin properties with new stage names. Instead a more comprehensive approach is needed to make sure we have a clear user flow during bulk copy/paste. Given the 6.2.2 timeline we are not adding a new user flow to 6.2.2 at this point.

### CODE REVIEW

**Review by elfenheart — APPROVED** [review]
👍  LGTM
